app/services/utils.ts

```diff
@@ -221,4 +221,14 @@ export default class Utils {
   static sleep(ms: number): Promise<void> {
     return new Promise(r => setTimeout(r, ms));
   }
+
+  static getReadableFileSizeString(fileSizeInBytes: number): string {
+    let i = -1;
+    const byteUnits = [' kB', ' MB', ' GB', ' TB', 'PB', 'EB', 'ZB', 'YB'];
+    do {
+      fileSizeInBytes = fileSizeInBytes / 1024;
+      i++;
+    } while (fileSizeInBytes > 1024);
+    return Math.max(fileSizeInBytes, 0.1).toFixed(1) + byteUnits[i];
```

Comment:
What if `i >= byteUnits.length`?

Reply:
while this is technically unclamped i think the probability of someone making and successfully uploading a 1025+YB file is extremely improbable, and the only place we use it has a max filesize of 2MB